### PR TITLE
Refactor Compare Table

### PR DESCRIPTION
**Pull request by jakelauer27**
Refactored the grid components so that I could add a highlight to currently selected rows.

Now the table headers are dynamically created and so are the columns for each character.

To do this I had to add a third data item in data.js. It's a cleaned version of only the properties I needed for the compare table.

### CODE REVIEW

File: src/Compare.js [hunk 1]
```diff
@@ -81,37 +84,18 @@ class Compare extends Component {
...
-            <th onClick={e => this.sortTable(e, 'speeds', 'fast_fall_speed')} 
-                className='up table-header-fast-fall'>Fst Fl Spd</th>
-            <th onClick={e => this.sortTable(e, 'counter')} 
-                className='up table-header-counter'>Ctr</th>
-            <th onClick={e => this.sortTable(e, 'strongest_smash', 'attack')} 
-                className='up table-header-smash'>Strongest Smash</th>
-            <th onClick={e => this.sortTable(e, 'strongest_smash', 'damage')} 
-                className='up table-header-smash-damage'>Smash Dmg</th>
+            {
+             categories.map( category => {
```
Comment (JustinTorrance) [inline src/Compare.js]: Very dry. No suggested edits.

File: src/CompareItem.js [hunk 2]
```diff
@@ -1,30 +1,37 @@
...
-        <td className={props.row}>{props.character.speeds.air_speed}</td>
-        <td className={props.row}>{props.character.speeds.fall_speed}</td>
-        <td className={props.row}>{props.character.speeds.fast_fall_speed}</td>
-        <td className={props.row}>{props.counter}</td>
-        <td className={props.row}>{props.character.strongest_smash.attack}</td>
-        <td className={props.row}>{props.character.strongest_smash.damage}</td>
+        <td className={this.props.row}><img src={this.props.character.images.small_icon}/></td>
+        {
+          categories.map( category => {
+            let highlight = '';
```
Comment (JustinTorrance) [inline src/CompareItem.js]: This is fire.

File: src/Compare.js [hunk 3]
```diff
@@ -1,25 +1,28 @@
...
 import './styles/App.css';
 
+
 class Compare extends Component {
   constructor() {
     super();
     this.state = {
-      data: characters.slice(0, characters.length)
+      data: characters.slice(0, characters.length),
+      highlight: 'Name'
```
Comment (libbyeh) [inline src/Compare.js]: Ah, interesting adding the highlight in the constructor. No edit, just wanted to comment!

File: src/styles/App.css [hunk 4]
```diff
@@ -61,13 +70,12 @@ table, th, tr, td {
...
 }
 body {
-  background: #929292;
-  background: -webkit-linear-gradient(to right, #494949, #1f1f1f);
-  background: linear-gradient(to right, #a3a3a3, #505050);
-}
-
-h1 {
-  color: blue;
+  background-image: url("../images/night_battlefield.jpg");
```
Comment (libbyeh) [inline src/styles/App.css]: Love the background overlay. Lets discuss if we want the same overlay on the other pages or change it up?

File: src/styles/_compare.scss [hunk 5]
```diff
@@ -48,3 +48,12 @@ table, th, tr, td {
     padding: 0px 8px;
   }
 }
+
+.arrow {
```
Comment (libbyeh) [inline src/styles/_compare.scss]: Glad you added the arrow. I think we should use the same one on the Character and Stage filters for consistency.

**Pull request merged**